packages/api/src/external/commonwell/patient.ts

```diff
@@ -209,36 +218,72 @@ export async function registerAndLinkPatientInCW(
       commonwellPatient,
       commonwellPatientId,
       patientRefLink,
-      storeIdsAndStatus,
+      patient,
       getOrgIdExcludeList,
     });
 
-    if (requestId) {
-      const startedAt = patient.data.patientDiscovery?.startedAt;
-
-      analytics({
-        distinctId: patient.cxId,
-        event: EventTypes.patientDiscovery,
-        properties: {
-          hie: MedicalDataSource.COMMONWELL,
-          patientId: patient.id,
-          requestId,
-          pdLinks: networkLinks?.length ?? 0,
-          duration: elapsedTimeFromNow(startedAt),
-        },
+    let cwLinks: NetworkLink[] = [];
+    if (networkLinks) {
+      cwLinks = await createCwLinks(patient, networkLinks);
+    }
+
+    analytics({
+      distinctId: patient.cxId,
+      event: EventTypes.patientDiscovery,
+      properties: {
+        hie: MedicalDataSource.COMMONWELL,
+        patientId: patient.id,
+        requestId,
+        pdLinks: networkLinks?.length ?? 0,
+        duration: elapsedTimeFromNow(startedAt),
+      },
+    });
+
+    let foundNewDemographics = false;
+    if (rerunPdOnNewDemographics) {
+      foundNewDemographics = checkForNewDemographics(patient, cwLinks);
+    }
+
+    let scheduledPdRequest = getCWData(patient.data.externalData)?.scheduledPdRequest;
+    if (foundNewDemographics && rerunPdOnNewDemographics) {
```

Comment:
`foundNewDemographics` can only be true if `rerunPdOnNewDemographics` is true so no need for extra check here

Reply:
Good call. Fixed.